apps/air-discount-scheme/backend/src/app/modules/discount/discount.service.ts

```diff
@@ -63,13 +71,90 @@ export class DiscountService {
     return this.cacheManager.get(cacheId)
   }
 
-  async createDiscountCode(nationalId: string): Promise<Discount> {
+  async createDiscountCode(
+    nationalId: string,
+    connectableFlights: Flight[],
+  ): Promise<Discount> {
     const discountCode = this.generateDiscountCode()
     const cacheId = CACHE_KEYS.discount(uuid())
-    await this.setCache<CachedDiscount>(cacheId, { nationalId, discountCode })
+
+    let connectionDiscountCodes: ConnectionDiscountCode[] = []
+
+    const previousCacheId = CACHE_KEYS.user(nationalId)
+    const previousCache = await this.getCache<CachedDiscount>(previousCacheId)
+
+    let previousFlightIds: string[] = []
+
+    if (previousCache?.connectionDiscountCodes) {
+      connectionDiscountCodes = previousCache.connectionDiscountCodes
+      previousFlightIds = connectionDiscountCodes.map((cdc) => cdc.flightId)
+    }
+
+    const connectableFlightCounts = connectableFlights.length
```

Comment:
If I am not mistaken, this could be simplified a lot by creating the connection discountCode when the flight is created.
So the POST /flights endpoint would check if the flight is connectable and if so it would call the discountService and create the connectionDiscountCode for that flight.

Then we would need all the checks here

Reply:
This is something that should be looked at in phase 2